prototype2/common/Producer.cpp

```diff
@@ -46,68 +35,60 @@ void Producer::event_cb(RdKafka::Event &event) {
 }
 
 ///
-Producer::Producer(std::string broker, std::string topicstr) :
-  ProducerBase(), topicString(topicstr) {
+Producer::Producer(std::string Broker, std::string Topic) :
+  ProducerBase(), TopicName(Topic) {
 
-  conf = RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL);
-  tconf = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
+  Config.reset(RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL));
+  TopicConfig.reset(RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC));
 
-  if (conf == nullptr) {
+  if (Config == nullptr) {
```

Comment:
For `unique_ptr`, I think it would be more idiomatic to use `operator bool` to check validity, i.e.:
https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool

Reply:
I do not agree with you as the comparison makes it clear to the reader that `Config` is a pointer and not a boolean or something else. Furthermore, I only changed the name here. The comparison is what is currently used.

With that said, the [C++ guidelines](https://isocpp.github.io/CppCoreGuidelines/) ([ES.87](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-if)) also suggests that the comparison is redundant and should not be used. I guess that if you feel strongly about it, we can change it. Do you have any additional comments @mortenjc ?

Reply:
No strong opinion.

Reply:
I prefer things more terse in general, but in this case I don't care enough,